### PR TITLE
[iOS] Add support for sharing ntp background image service

### PR DESCRIPTION
**Pull request by kylehickinson**
Accompanying brave-ios branch: https://github.com/brave/brave-ios/compare/brave-core-1.54.x...ntp-component-updater

Resolves https://github.com/brave/brave-browser/issues/30871

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/ntp_background_images/browser/sponsored_images_component_data.cc [hunk 1]
```diff
@@ -27,7 +27,8 @@ constexpr SponsoredImagesComponentData kDemoData = {
 // This list should be synced with the list of generateNTPSponsoredImages.js
 // and packageNTPSponsoredImagesComponents.js in brave-core-crx-packager.
 constexpr SponsoredImagesComponentData kRegionalData[] = {
-#if BUILDFLAG(IS_ANDROID)
+// TODO: iOS needs its own designated set of component identifiers
+#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS)
```
Comment (kylehickinson) [inline components/ntp_background_images/browser/sponsored_images_component_data.cc:31]: These need to be swapped for a new set of component IDs for iOS but the rest of the PR is reviewable

Comment (kylehickinson) [reply components/ntp_background_images/browser/sponsored_images_component_data.cc]: Changes needed are here: https://github.com/brave/brave-core/pull/18706

Will rebase once merged

File: components/ntp_background_images/browser/ntp_background_images_data.h [hunk 2]
```diff
@@ -12,10 +12,13 @@
...
 #include "base/values.h"
 #include "third_party/abseil-cpp/absl/types/optional.h"
-#include "third_party/skia/include/core/SkColor.h"
 #include "ui/gfx/geometry/point.h"
 #include "ui/gfx/geometry/rect.h"
 
+#if !BUILDFLAG(IS_IOS)
+#include "third_party/skia/include/core/SkColor.h"
+#endif
+
```
Comment (simonhong) [inline components/ntp_background_images/browser/ntp_background_images_data.h:21]: optional - I realized that this header including is not necessary for desktop also. If you want, you can just delete this.

File: components/ntp_background_images/browser/ntp_sponsored_images_data.h [hunk 3]
```diff
@@ -12,10 +12,13 @@
...
 #include "base/values.h"
 #include "third_party/abseil-cpp/absl/types/optional.h"
-#include "third_party/skia/include/core/SkColor.h"
 #include "ui/gfx/geometry/point.h"
 #include "ui/gfx/geometry/rect.h"
 
+#if !BUILDFLAG(IS_IOS)
+#include "third_party/skia/include/core/SkColor.h"
+#endif
+
```
Comment (simonhong) [inline components/ntp_background_images/browser/ntp_sponsored_images_data.h:21]: dotto

**Review by simonhong — APPROVED** [review]
components/ntp_background_images ++

File: components/ntp_background_images/browser/ntp_background_images_data.h (hunk 2 above)
Comment (kylehickinson) [reply components/ntp_background_images/browser/ntp_background_images_data.h:21]: Sounds good, will remove when rebasing after https://github.com/brave/brave-core/pull/18706 merges as that will also include a few changes :)

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.54.65 Chromium: 114.0.5735.133 (Official Build) nightly (64-bit)
-- | --
Revision | fbfa2ce68d01b2201d8c667c2e73f648a61c4f4a-refs/branch-heads/5735@{#1270}
OS | Windows 11 Version 22H2 (Build 22621.1848)
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/30871#issue-1744611079, ensured that both NTP & NTP SI images were being displayed/working as expected. This will also be easily verified by default once the change lands in `1.53.x` and QA starts running through verifications with the new build(s).

Verification **PASSED** on `Pixel 6` running `Android 14` using the following build(s):

```
Brave | 1.54.66 Chromium: 114.0.5735.133 (Official Build) canary (64-bit)
--- | ---
Revision | fbfa2ce68d01b2201d8c667c2e73f648a61c4f4a-refs/branch-heads/5735@{#1270}
OS | Android 14; Build/UPB3.230519.008; 34; REL
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/30871#issue-1744611079, ensured that both NTP & NTP SI images were being displayed/working as expected. This will also be easily verified by default once the change lands in `1.53.x` and QA starts running through verifications with the new build(s).

**Pull request merged**